mingw_files/nrnmingwenv.sh

```diff
@@ -1,53 +1,99 @@
-#!/bin/sh
+#!/bin/bash
+
 set -ex
 
+# Primarily (autotools and cmake) copies minimal developent files
+# to allow nrnivmodl (mknrndll) to build nrnmech.dll .
+# The -cmake second arg option does a few extra things to prepare
+# for making the setup.exe. I.e. remove some inadvertenty installed
+# file that needlessly bloat up the setup.exe, make sure that
+# dlls needed by nrniv.exe, etc, are present, copy some mswin
+# specific scripts, and make sure the neurondemo nrnmech.dll
+# is present.
+
 # if arg then assume it is destination, eg, c:/marshalnrn64/nrn
-if test $# = 1 ; then
-  N="$1/mingw"
+if test $# -ge 1 ; then
+  N="$1"
 else
-  N='c:/nrn/mingw'
-  rm -r -f "$N"
+  N='c:/nrn'
+  rm -r -f "$N/mingw"
+fi
+
+NM=$N/mingw
+
+if test "$2" = "-cmake" ; then
+  is_cmake=yes
+  srcdir="$3"
+fi
+
+if test "$is_cmake" = "yes" ; then
+  # Minimal runtime msmpi (assumes installed in c:/ms-mpi)
+  mpiinst=c:/ms-mpi
```

Comment:
In future I assume these scripts can be driven via configure_file to populate correct paths (instead of hard-coding).

Reply:
Yes. Slightly less convenient at the moment since this is not a configure_file in autotools context. And setting it here is minutes faster on a virtualbox guest as it does not require the cmake;make;make install; make setup_exe but just the last.  On the otherhand, I believe cmake has the path already figured out. This should be eventually turned into a separate configure file and nrnmingwenv.sh return to its originial form used solely to copy the relevant
developer files needed by the mknrndll (nrnivmodl).  An issue that is related is the  fact that
this all depends on a prior 'make install' and the  dependencies of this process of marshaling  in preparation for creating the setup.exe  is not at all specified to cmake in a way that avoids duplicating many actions that may have already been accomplished in a previous  'make setup_exe'